build-tools/packages/build-tools/src/fluidBuild/buildGraph.ts

```diff
@@ -514,12 +514,17 @@ export class BuildGraph {
 	}
 
 	private async isUpToDate() {
-		const isUpToDateP = new Array<Promise<boolean>>();
-		this.buildPackages.forEach((node) => {
-			isUpToDateP.push(node.isUpToDate());
-		});
-		const isUpToDateArr = await Promise.all(isUpToDateP);
-		return isUpToDateArr.every((isUpToDate) => isUpToDate);
+		try {
+			const isUpToDateP = new Array<Promise<boolean>>();
+			this.buildPackages.forEach((node) => {
+				isUpToDateP.push(node.isUpToDate());
+			});
+			const isUpToDateArr = await Promise.all(isUpToDateP);
+			return isUpToDateArr.every((isUpToDate) => isUpToDate);
+		} catch {
+			// If checking the up-to-date state fails, we assume that the build is not up to date.
+			return false;
```

Comment:
We should log the exception and stack if possible.

Reply:
maybe in verbose or debug trace?

Reply:
Yeah, verbose.debug would be fine.

Reply:
Note that the value of this is just to short circuit the build.   So, it is ok to just return false, and let the task individually error out later.